### PR TITLE
Adjust json parser to new format

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Adjust parser to new format

### CODE REVIEW

**Comment by pb82** [comment]
@wtrocki why two examples files? (`assets/mobile-services.json` and `assets/dummy-mobile-services.json`)

**Comment by wtrocki** [comment]
@secondsun Can we say why we have 2 example files? They both are used for unit test purposes.

**Comment by secondsun** [comment]
They are used for unit tests.

File: core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonParser.java [hunk 1]
```diff
@@ -48,44 +48,22 @@ private void parseMobileCoreArray(final JSONArray array) throws JSONException, I
...
-            String name = namesArray.getString(i);
-            switch (name) {
-                case "type":
-                    serviceConfigBuilder.setType(config.getString("type"));
-                    break;
-                case "uri":
-                    serviceConfigBuilder.setUri(config.getString("uri"));
-                    break;
-                case "headers":
-                    addHeaders(serviceConfigBuilder, config.getJSONObject("headers"));
```
Comment (secondsun) [inline core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonParser.java]: Are we intentionally removing headers?

File: core/src/debug/assets/dummy-mobile-services.json [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
+            }
+        },
+        {
+            "id": "keycloak",
+            "name": "keycloak",
+            "type": "keycloak",
+            "url": "https://www.mocky.io/v2/5a6b59fb31000088191b8ac6",
+            "config": {
+                "auth-server-url": "https://keycloak-myproject.192.168.64.74.nip.io/auth",
+                "clientId": "juYAlRlhTyYYmOyszFa",
```
Comment (TommyJ1994) [inline core/src/debug/assets/dummy-mobile-services.json]: I don't think we need this? Keycloak uses `resource` as defined below for specifying the clientID.

File: core/src/debug/assets/mobile-services.json [hunk 3]
```diff
@@ -1,26 +1,37 @@
...
+            "id": "keycloak",
+            "name": "keycloak",
+            "type": "keycloak",
+            "url": "https://www.mocky.io/v2/5a6b59fb31000088191b8ac6",
             "config": {
-                "auth-server-url": "http://keycloak-myproject.192.168.37.1.nip.io/auth",
-                "clientId": "OeFYkwEuUKXvDsSudfXc",
-                "headers": {},
+                "auth-server-url": "https://keycloak-myproject.192.168.64.74.nip.io/auth",
+                "clientId": "juYAlRlhTyYYmOyszFa",
```
Comment (TommyJ1994) [inline core/src/debug/assets/mobile-services.json]: Same here, can remove this.

**Comment by danielpassos** [comment]
@pb82 I have added 2 different JSON conf file to test the [`setConfigFileName`](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/MobileCore.java#L214-L217). See [MobileCoreTest.java](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/test/java/org/aerogear/mobile/core/MobileCoreTest.java#L51-L63)

**Comment by pb82** [comment]
@secondsun @danielpassos thanks, was only curious.

**Pull request merged**